src/code/z_player_lib.c

```diff
@@ -1798,49 +1987,399 @@ void Player_DrawZoraShield(PlayState* play, Player* player) {
+                    func_80125340();
+                    func_80125318(pos, rot);
+
+                    if (bubble != NULL) {
+                        s32 requiredScopeTemp[2];
+
+                        player->unk_AF0[0].x = 1.0f - (bubble->bubble.unk_144 * 0.03f);
+                        player->unk_AF0[0].y = 1.0f - (bubble->bubble.unk_144 * 0.01f);
+                        player->unk_AF0[0].z = 1.0f - (bubble->bubble.unk_144 * 0.04f);
+                        rot->z = bubble->bubble.unk_144 * 320.0f;
+                    } else if (player->skelAnime.animation == &gPlayerAnim_pn_drinkend) {
+                        func_80124618(D_801C03E0, player->skelAnime.curFrame, player->unk_AF0);
+                    } else {
+                        func_80124618(D_801C03C0, player->unk_284.curFrame, player->unk_AF0);
+                    }
+
+                    Matrix_Scale(player->unk_AF0[0].x, player->unk_AF0[0].y, player->unk_AF0[0].z, MTXMODE_APPLY);
+                }
+            }
+        } else if (limbIndex == PLAYER_LIMB_TORSO) {
+            if ((player->skelAnime.animation == &gPlayerAnim_pg_gakkistart) ||
+                (player->skelAnime.animation == &gPlayerAnim_pg_wait) ||
+                (player->skelAnime.animation == &gPlayerAnim_pg_punchC) ||
+                (player->transformation == PLAYER_FORM_ZORA)) {
+                Matrix_TranslateRotateZYX(pos, rot);
+                if (player->transformation == PLAYER_FORM_GORON) {
+                    func_80125340();
+                }
+
+                func_80125318(pos, rot);
+                if ((player->transformation != PLAYER_FORM_ZORA) ||
+                    (player->skelAnime.animation == &gPlayerAnim_pz_wait)) {
+                    struct_80124618* phi_a0 =
+                        (player->skelAnime.animation == &gPlayerAnim_pz_wait)
+                            ? D_801C0608
+                            : ((player->skelAnime.animation == &gPlayerAnim_pg_wait)
+                                   ? D_801C0628
+                                   : ((player->skelAnime.animation == &gPlayerAnim_pg_punchC) ? D_801C07C0
+                                                                                              : D_801C0460));
+
+                    func_80124618(phi_a0, player->skelAnime.curFrame, player->unk_AF0);
+                } else {
+                    player->unk_AF0[0].y = 1.0f;
+                }
+
+                if (player->transformation == PLAYER_FORM_ZORA) {
+                    player->unk_AF0[0].y *= 1.0f + (0.29999995f * player->unk_B10[0]);
+                    player->unk_AF0[0].x = 1.0f;
+                    player->unk_AF0[0].z = 1.0f;
+                }
+                Matrix_Scale(player->unk_AF0[0].x, player->unk_AF0[0].y, player->unk_AF0[0].z, MTXMODE_APPLY);
+            }
+        } else if (limbIndex == PLAYER_LIMB_UPPER_ROOT) {
+            s16 xRot;
+            s16 zRot = 0x44C;
```

Comment:
```suggestion
            s16 rotX;
            s16 rotZ = 0x44C;
```
I think this order is much more common?